support-macros/src/lib.rs

```diff
@@ -0,0 +1,58 @@
+#![recursion_limit = "128"]
+extern crate proc_macro;
+extern crate proc_macro2;
+
+use proc_macro::TokenStream;
+
+use quote::quote;
+use syn;
+use syn::Error;
+use syn::ItemFn;
+
+#[proc_macro_attribute]
+pub fn libtock_main(_: TokenStream, input: TokenStream) -> TokenStream {
+    generate_main_wrapped(input.into()).into()
+}
+
+fn generate_main_wrapped(input: proc_macro2::TokenStream) -> proc_macro2::TokenStream {
+    try_generate_main_wrapped(input).unwrap_or_else(|err| err.to_compile_error().into())
+}
+
+fn try_generate_main_wrapped(
+    input: proc_macro2::TokenStream,
+) -> Result<proc_macro2::TokenStream, Error> {
+    let ast = syn::parse2::<ItemFn>(input)?;
+    let block = ast.block;
+    Ok(quote!(
+        fn main() -> ::libtock::result::TockResult<()> {
```

Comment:
The macro or the main function should be callable exactly once. @jrvanwhy added a runtime check and I think this is the only possible option.

Reply:
The macro will be callable multiple times but this will lead to compile errors. Having multiple main functions will also lead to compile errors.

Reply:
The problem is that main() can recurse into main(), causing undefined behavior.

Reply:
I took your solution and added it to the macro.